<!DOCTYPE html>
<html lang="en">
<head>
<title>Analyzing RNA-seq Data with DESeq2 (Redesigned)</title>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="https://www.w3schools.com/w3css/4/w3.css">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
<link href='https://fonts.googleapis.com/css?family=Montserrat' rel='stylesheet'>
<style>
body {font-family: 'Montserrat', sans-serif}
h1,h2,h3,h4,h5,h6 {font-family:'Montserrat'}
</style>
</head>
<body>

<!-- Sidebar/menu -->
<nav class="w3-sidebar w3-light-grey w3-collapse w3-top w3-padding" style="z-index:3;width:30%;min-width:300px;" id="mySidebar"><br>
  <div class="w3-container w3-center" style="margin-top:8%; margin-left:8%;margin-right:8%">
    <img src="bioconductor_o.png" style="width:35%;max-width:100px;" alt="Bioconductor Logo">
  </div>
  <div class="w3-container w3-center" style="margin-left:8%;margin-right:8%;font-weight:bolder">
    <p class="w3-large">Analyzing RNA-seq Data with DESeq2</p>
    <p class="w3-small">Michael I. Love, Simon Anders, and Wolfgang Huber <br> Last updated: 08/25/2022 </p>
  </div>
  <div class="w3-container w3-small" style="margin-left:8%;margin-right:8%">
    <p>if you use DESeq2 in published research, please cite:</p>
    <ul class="w3-leftbar w3-theme-border" style="list-style:none">
        <li>Love, M.I., Huber, W., Anders, S. (2014) Moderated estimation of fold change and 
            dispersion for RNA-seq data with DESeq2. Genome Biology, 15:550. 10.1186/s13059-014-0550-8
    </ul>
  </div>
  <div class="w3-container w3-small" style="margin-left:8%;margin-right:8%">
    <p>Other Bioconductor packages with similar aims are edgeR, limma, DSS, EBSeq, and baySeq.</p>
  </div>
  <div class="w3-bar-block" style="margin-top:4%;margin-left:8%;margin-right:8%;font-weight:bold">
    <a href="#abstract" onclick="w3_close()" class="w3-bar-item w3-button w3-hover-blue">Abstract</a> 
    <div>
        <a href="javascript:void(0)" onclick="myAccordion('standard workflow')" class="w3-bar-item w3-button w3-hover-blue">Standard Workflow <i class="fa fa-caret-down"></i></a>
        <div id="standard workflow" class="w3-hide">
          <a href="#quickstart" class="w3-bar-item w3-button w3-hover-lime w3-small" style="font-weight:normal">&nbsp&nbsp Quick start</a>
          <a href="#help" class="w3-bar-item w3-button w3-hover-lime w3-small" style="font-weight:normal">&nbsp&nbsp How to get help for DESeq2</a>
          <a href="javascript:void(0)" onclick="myAccordion('standard workflow input data')" class="w3-bar-item w3-button w3-hover-lime w3-small" style="font-weight:normal">&nbsp&nbsp Input data <i class="fa fa-caret-down"></i></a>
          <div id="standard workflow input data" class="w3-hide">
            <a href="#why" class="w3-bar-item w3-button w3-hover-khaki w3-small" style="font-weight:normal">&nbsp&nbsp&nbsp&nbsp Why un-normalized counts?</a>
            <a href="#deseqdataset" class="w3-bar-item w3-button w3-hover-khaki w3-small" style="font-weight:normal">&nbsp&nbsp&nbsp&nbsp The DESeqDataSet</a>
          </div>
        </div>
    </div>
    <a href="#datatransformations" onclick="w3_close()" class="w3-bar-item w3-button w3-hover-blue">Data Transformations</a> 
    <div>
        <a href="javascript:void(0)" onclick="myAccordion('variations')" class="w3-bar-item w3-button w3-hover-blue">Variations <i class="fa fa-caret-down"></i></a>
        <div id="variations" class="w3-hide">
          <a href="#waldtestindividualsteps" class="w3-bar-item w3-button w3-hover-lime w3-small" style="font-weight:normal">&nbsp&nbsp Wald test individual steps</a>
          <a href="#contrasts" class="w3-bar-item w3-button w3-hover-lime w3-small" style="font-weight:normal">&nbsp&nbsp Contrasts</a>
        </div>
    </div>
    <a href="#theory" onclick="w3_close()" class="w3-bar-item w3-button w3-hover-blue">Theory Behind DESeq2</a> 
    <a href="#faq" onclick="w3_close()" class="w3-bar-item w3-button w3-hover-blue">Frequently Asked Questions</a> 
    <a href="#sessioninfo" onclick="w3_close()" class="w3-bar-item w3-button w3-hover-blue">Session Info</a>
    
  </div>
</nav>

<!-- Top menu on small screens -->
<header class="w3-container w3-top w3-hide-large w3-light-grey w3-small w3-padding">
  <a href="javascript:void(0)" class="w3-button w3-light-grey w3-margin-right" onclick="w3_open()">☰</a>
  <span>Analyzing RNA-seq data with DESeq2</span>
</header>

<!-- Overlay effect when opening sidebar on small screens -->
<div class="w3-overlay w3-hide-large" onclick="w3_close()" style="cursor:pointer" title="close side menu" id="myOverlay"></div>

<!-- !PAGE CONTENT! -->
<div class="w3-main" style="margin-left:35%;margin-right:5%">

  <!-- Abstract -->
    <div class="w3-container" style="margin-top:80px" id="abstract">
        <h1 class="w3-xxlarge"><b>Abstract</b></h1>
        <hr style="width:50px;border:5px solid #1a81c2" class="w3-round">
        <p>A basic task in the analysis of count data from RNA-seq is the detection of differentially expressed genes. 
            The count data are presented as a table which reports, for each sample, the number of sequence fragments that 
            have been assigned to each gene. Analogous data also arise for other assay types, including comparative ChIP-Seq, 
            HiC, shRNA screening, and mass spectrometry. An important analysis question is the quantification and statistical 
            inference of systematic changes between conditions, as compared to within-condition variability. The package DESeq2 
            provides methods to test for differential expression by use of negative binomial generalized linear models; the 
            estimates of dispersion and logarithmic fold changes incorporate data-driven prior distributions. This vignette 
            explains the use of the package and demonstrates typical workflows. An RNA-seq workflow on the Bioconductor website 
            covers similar material to this vignette but at a slower pace, including the generation of count matrices from FASTQ 
            files. DESeq2 package version: 1.37.6
        </p>
    </div>

  <!-- Standard Workflow -->
    <div class="w3-container" style="margin-top:80px">
        <h1 class="w3-xxlarge"><b>Standard Workflow</b></h1>
        <hr style="width:50px;border:5px solid #1a81c2" class="w3-round">
        <div class="w3-container" style="margin-top:30px" id="quickstart">
            <h1 class="w3-xlarge"><b>Quick start</b></h1>
            <hr style="width:30px;border:5px solid #87b13f" class="w3-round">
            <p>Here we show the most basic steps for a differential expression analysis. There are a variety of steps upstream 
                of DESeq2 that result in the generation of counts or estimated counts for each sample, which we will discuss in 
                the sections below. This code chunk assumes that you have a count matrix called <code class="w3-codespan">cts</code> and a table of sample information 
                called <code class="w3-codespan">coldata</code>. The <code class="w3-codespan">design</code> indicates how to model the samples, here, that we want to 
                measure the effect of the <code class="w3-codespan">condition</code>, controlling for <code class="w3-codespan">batch</code> differences. The two factor 
                variables batch and condition should be columns of <code class="w3-codespan">coldata</code>.
            </p>
            <div class="w3-code notranslate jsHigh w3-card w3-border-red">
                dds &lt- DESeqDataSetFromMatrix(countData = cts, colData = coldata, design= ~ batch + condition)<br>
                dds &lt- DESeq(dds)<br>
                resultsNames(dds) # lists the coefficients<br>
                res &lt- results(dds, name="condition_trt_vs_untrt")<br>
                # or to shrink log fold changes association with condition:<br>
                res &lt- lfcShrink(dds, coef="condition_trt_vs_untrt", type="apeglm")<br>
            </div>
            <p>The following starting functions will be explained below:
                <li>If you have performed transcript quantification (with Salmon, kallisto, RSEM, etc.) you could import the data with tximport, which produces a list, 
                    and then you can use <code class="w3-codespan">DESeqDataSetFromTximport()</code>.
                </li>
                <li>If you imported quantification data with tximeta, which produces a SummarizedExperiment with additional metadata, you can then use <code class="w3-codespan">DESeqDataSet()</code>.
                </li>
                <li>If you have htseq-count files, you can use <code class="w3-codespan">DESeqDataSetFromHTSeq()</code>.</li>
            </p>
        </div>
        <div class="w3-container" style="margin-top:30px" id="help">
            <h1 class="w3-xlarge"><b>How to get help for DESeq2</b></h1>
            <hr style="width:30px;border:5px solid #87b13f" class="w3-round">
            <p>Any and all DESeq2 questions should be posted to the Bioconductor support site, which serves as a searchable knowledge base of questions and answers:
                https://support.bioconductor.org</p>
                
            <p>Posting a question and tagging with “DESeq2” will automatically send an alert to the package authors to respond on the support site. See the first question in the list of Frequently Asked Questions (FAQ) for information about how to construct an informative post.
            </p>    
            <p>You should not email your question to the package authors, as we will just reply that the question should be posted to the Bioconductor support site.
            </p>
        </div>
        <div class="w3-container" style="margin-top:30px">
            <h1 class="w3-xlarge"><b>Input data</b></h1>
            <hr style="width:30px;border:5px solid #87b13f" class="w3-round">
            <div class="w3-container" style="margin-top:20px" id="why">
                <h1 class="w3-large"><b>Why un-normalized counts?</b></h1>
                <hr style="width:15px;border:5px solid #f1d855" class="w3-round">
                <p>Actual texts from the original website are skipped here.  
                </p>
            </div>
            <div class="w3-container" style="margin-top:20px" id="deseqdataset">
                <h1 class="w3-large"><b>The DESeqDataSet</b></h1>
                <hr style="width:15px;border:5px solid #f1d855" class="w3-round">
                <p>Actual texts from the original website are skipped here. Insert my favorate image in the tutorial to show how images on this website will be displayed:
                </p>
                <img src="PCA.png" style="width:70%;max-width:700px;" alt="PCA plot">
            </div>
        </div>
    </div>

    <!-- Data Transformations -->
    <div class="w3-container" style="margin-top:80px" id="datatransformations">
        <h1 class="w3-xxlarge"><b>Data Transformations</b></h1>
        <hr style="width:50px;border:5px solid #1a81c2" class="w3-round">
        <p>Actual texts from the original website are skipped here.
        </p>
    </div>

    <!-- Variations -->
    <div class="w3-container" style="margin-top:80px">
        <h1 class="w3-xxlarge"><b>Variations</b></h1>
        <hr style="width:50px;border:5px solid #1a81c2" class="w3-round">
        <div class="w3-container" style="margin-top:30px" id="waldtestindividualsteps">
            <h1 class="w3-xlarge"><b>Wald test individual steps</b></h1>
            <hr style="width:30px;border:5px solid #87b13f" class="w3-round">
            <p>Actual texts from the original website are skipped here.
        </div>
        <div class="w3-container" style="margin-top:30px" id="contrasts">
            <h1 class="w3-xlarge"><b>Contrasts</b></h1>
            <hr style="width:30px;border:5px solid #87b13f" class="w3-round">
            <p>Actual texts from the original website are skipped here.
        </div>
    </div>

    <!-- Theory Behind DESeq2 -->
    <div class="w3-container" style="margin-top:80px" id="theory">
        <h1 class="w3-xxlarge"><b>Theory Behind DESeq2</b></h1>
        <hr style="width:50px;border:5px solid #1a81c2" class="w3-round">
        <p>Actual texts from the original website are skipped here.
        </p>
    </div>

    <!-- Frequently Asked Questions -->
    <div class="w3-container" style="margin-top:80px" id="faq">
        <h1 class="w3-xxlarge"><b>Frequently Asked Questions</b></h1>
        <hr style="width:50px;border:5px solid #1a81c2" class="w3-round">
        <p>Actual texts from the original website are skipped here.
        </p>
    </div>

    <!-- Session Info -->
    <div class="w3-container" style="margin-top:80px; margin-bottom: 80px;" id="sessioninfo">
        <h1 class="w3-xxlarge"><b>Session Info</b></h1>
        <hr style="width:50px;border:5px solid #1a81c2" class="w3-round">
        <p>Actual texts from the original website are skipped here.
        </p>
    </div>

</div>

  

<!-- End page content -->

<script src="https://www.w3schools.com/lib/w3codecolor.js"></script>
<script>
w3CodeColor();

// Script to open and close sidebar
function w3_open() {
  document.getElementById("mySidebar").style.display = "block";
  document.getElementById("myOverlay").style.display = "block";
}
 
function w3_close() {
  document.getElementById("mySidebar").style.display = "none";
  document.getElementById("myOverlay").style.display = "none";
}

// Modal Image Gallery
function onClick(element) {
  document.getElementById("img01").src = element.src;
  document.getElementById("modal01").style.display = "block";
  var captionText = document.getElementById("caption");
  captionText.innerHTML = element.alt;
}

// Accordions
function myAccordion(id) {
  var x = document.getElementById(id);
  if (x.className.indexOf("w3-show") == -1) {
    x.className += " w3-show";
    x.previousElementSibling.className += " w3-theme";
  } else { 
    x.className = x.className.replace("w3-show", "");
    x.previousElementSibling.className = 
    x.previousElementSibling.className.replace(" w3-theme", "");
  }
}
</script>

</body>
</html>
